tests/unittests/benchmark/task/test_task_wrapper.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any, Dict, List
+
+import pytest
+from orion.benchmark.task.base import BaseTask
+from orion.benchmark.task.task_wrapper import FixTaskDimensionsWrapper, TaskWrapper
+from orion.core.io.space_builder import SpaceBuilder
+
+
+class DumbTask(BaseTask):
+    def __init__(self, max_trials: int, seed: int = None, a: int = 100, b: int = 10, c: int = 1):
+        super().__init__(max_trials=max_trials)
+        self.a = a
+        self.b = b
+        self.c = c
+        self.seed = seed
+
+    def call(self, x: Dict) -> List[Dict]:
+        y = self.a * x["a"] + self.b * x["b"] + self.c * x["c"]
+        return [dict(name="dumb_task", type="objective", value=y)]
+
+    def get_search_space(self) -> Dict[str, str]:
+        return {
+            "a": "uniform(0, 10, discrete=True)",
+            "b": "uniform(0, 10, discrete=True)",
+            "c": "uniform(0, 10, discrete=True)",
+        }
+
+    @property
+    def configuration(self) -> Dict[str, Any]:
+        return {
+            type(self).__qualname__: dict(
+                max_trials=self.max_trials, seed=self.seed, a=self.a, b=self.b, c=self.c,
+            )
+        }
+
+
+class TestFixDimensionsWrapper:
+    def test_init(self):
+
+        original_task = DumbTask(max_trials=10)
+        wrapped_task = FixTaskDimensionsWrapper(original_task, fixed_dims={"c": 1.0})
+
+        get_y = lambda results: results[0]["value"]
+
+        assert get_y(original_task({"a": 1, "b": 1, "c": 1})) == 111
+        assert get_y(wrapped_task({"a": 1, "b": 1, "c": 1})) == 111
+
+        assert get_y(original_task({"a": 1, "b": 1, "c": 3})) == 113
+        assert get_y(wrapped_task({"a": 1, "b": 1, "c": 3})) == 111
+
+    def test_fixed_dim_needs_to_be_part_of_space(self):
+        with pytest.raises(ValueError):
+            original_task = DumbTask(10)
+            wrapped_task = FixTaskDimensionsWrapper(original_task, fixed_dims={"foo": 1.0})
+
+    @pytest.mark.skip(reason="Not sure this makes sense. ")
+    def test_space_samples_also_have_fixed_value(self):
```

Comment:
The wrapped task should be passed points containing the fixed dimensions. However the wrapped should receive points only containing the non-fix dimensions.

Reply:
So I would test the call, not the space sample.

Reply:
Not necessarily, right? If the wrapper fixes a value for a given dimension, say `{"foo": 1.0}`, then the wrapper receives `{"foo": 1.23}`, it then passes down a `{"foo": 1.0}` to the wrapped task!

Reply:
Why would you pass a point with dim `foo` if the space does not contain `foo`? From the exterior, the task does not have a `foo` dim (even though if we dig deep down we will find a wrapped task with a `foo` dim) I think this adaptation is solely the responsibility of the EVC or the knowledge base.

Reply:
> The wrapped task should be passed points containing the fixed dimensions. However the **wrapped** should receive points only containing the non-fix dimensions.

My confusion came from the "wrapped" here being used twice! You're right.

Reply:
Removed the test.

Reply:
oh! Sorry about the typo! :sweat_smile: